### ISSUE TITLE
1.11.0 - terraform provider `mongodbatlas_projects` access denied 

### ISSUE THREAD

**Issue by neo-eddie-nazarov** [mongodb/terraform-provider-mongodbatlas#1371]
Hi,

Using read only API keys for terraform plans, since provider version upgrade started getting the following error:

```
1, in data "mongodbatlas_projects" "projects":
   1: data "mongodbatlas_projects" "projects" {

Error getting project's api keys (<>): GET
https://cloud.mongodb.com/api/atlas/v1.0/groups/<>/apiKeys:
401 (request "USER_UNAUTHORIZED") Current user is not authorized to perform
this action.
```

Works just fine on 1.10.0

There was a previous issue here - https://github.com/mongodb/terraform-provider-mongodbatlas/issues/659 in case of its a regression bug. 

### Terraform CLI and Terraform MongoDB Atlas Provider Version
terraform 1.5.3
MongoDB provider 1.11.0

**Comment by Zuhairahmed** [comment]
hi @neo-eddie-nazarov can you share more of your terraform script with us so we can better investigate issue? also anything else you think we need to reproduce issue on our side. thank you

**Comment by neo-eddie-nazarov** [comment]
hey @Zuhairahmed,

Just simple data resource which is referenced later on. 
```
data "mongodbatlas_projects" "projects" {
  items_per_page = 500
}
```

credentials are passed using the environment variables:

```
MONGODB_ATLAS_PUBLIC_KEY
MONGODB_ATLAS_PRIVATE_KEY
```

The failure as described comes with the following error: 

```
Error getting project's api keys (<>): GET
https://cloud.mongodb.com/api/atlas/v1.0/groups/<>/apiKeys:
401 (request "USER_UNAUTHORIZED") Current user is not authorized to perform
this action.

```

this is working perfectly fine with `1.10.0` provider version but fails on `1.11.0` 

Currently pinned the provider to 1.10.0 to resolve this: 

```
    mongodbatlas = {
      source  = "mongodb/mongodbatlas"
      version = "1.10.0"
    }
```

**Comment by andreaangiolillo** [comment]
Hi @neo-eddie-nazarov,
Thanks for adding additional details to the issue. Could you also list the roles associated with your API keys?

Thanks

**Comment by neo-eddie-nazarov** [comment]
@andreaangiolillo its `Organization Read Only`

**Comment by andreaangiolillo** [comment]
Hi @neo-eddie-nazarov, 
I was able to reproduce the issue. However, I am getting a warning message and the data source is still stored in the tf state file without API keys info. Using `v1.10.0`, `1.10.1` and `1.10.2` I did not get the warning message but the data source was still missing the API key details. 

Could you confirm that when running `terraform state show data.mongodbatlas_projects.projects` using `v1.10.0` the `api_keys` field is not populated? 


```bash
╷
│ Warning: Error getting project's api keys
│
│   with data.mongodbatlas_projects.projects,
│   on main.tf line 577, in data "mongodbatlas_projects" "projects":
│  577: data "mongodbatlas_projects" "projects" {
│
│ Error getting project's api keys (64b8d3219c4cdc3f9108e318): GET https://cloud.mongodb.com/api/atlas/v1.0/groups/64b8d3219c4cdc3f9108e318/apiKeys: 401 (request "USER_UNAUTHORIZED") Current user is not authorized to perform
│ this action.
│
│ (and 16 more similar warnings elsewhere)
╵

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

Thanks

**Comment by Zuhairahmed** [comment]
@neo-eddie-nazarov confirming this is intended behavior so will Close this issue. The `Organization Read Only` role: 
provides read-only access to everything in the organization, including all projects in the organization. However this is not sufficient privilege to Create and Assign One Organization API Key to One Project. Instead suggest use higher role linked to your Programmatic API keys to access this data source. Prior version of our Terraform Provider had a bug which has since been corrected in v1.10.0 which explains why you are now seeing this "Current user is not authorized to perform this action" message. Hope this helps. 

Feel free to open a new issue if you need anything else. Thank you.
